storage/index/types.go

```diff
@@ -23,8 +23,27 @@ package index
 import (
 	"time"
 
+	"github.com/m3db/m3db/clock"
+	"github.com/m3db/m3db/x/xpool"
 	"github.com/m3db/m3ninx/index/segment"
+	"github.com/m3db/m3ninx/index/segment/mem"
 	"github.com/m3db/m3x/ident"
+	"github.com/m3db/m3x/instrument"
+)
+
+var (
+	// ReservedFieldNameID is the field name used to index the ID in the
+	// m3ninx subsytem.
+	ReservedFieldNameID = []byte("_m3db-id")
```

Comment:
nit: Can we use the same special character here, i.e. `_m3db_id`?  Just when debugging things typing "_m3db_id" over and over again is much easier than "_m3db-id" (I've noticed this when debugging strange JSON structures, thank you human brain...).

Reply:
haha sure thing